-1

Tested at: 6db77f215703f5e9dc7f076262293ba1b8861e55

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17838/18675/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>Copying tmp/slc6_amd64_gcc530/src/DQM/SiStripMonitorClient/bin/modulediff/modulediff to productstore area:
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQMServices/StreamerIO/plugins/DQMFileIterator.cc 
>> Building binary DQMCollector
Copying tmp/slc6_amd64_gcc530/src/DQMServices/Core/bin/DQMCollector/DQMCollector to productstore area:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/src/SiStripDcsInfo.cc: In member function 'void SiStripDcsInfo::addBadModules()':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/src/SiStripDcsInfo.cc:338:66: error: conversion from 'std::map<unsigned int, unsigned int>' to non-scalar type 'std::map<unsigned int, short unsigned int>' requested
     std::map<uint32_t,uint16_t> lumiCountBadModules = it->second.NLumiDetectorIsFaulty;
                                                                  ^
>> Building shared library tmp/slc6_amd64_gcc530/src/DQMServices/Core/src/DQMServicesCore/libDQMServicesCore.so
Copying tmp/slc6_amd64_gcc530/src/DQMServices/Core/src/DQMServicesCore/libDQMServicesCore.so to productstore area:
Leaving library rule at DQMServices/Core
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 64 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/bin/listbadmodule.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/bin/check_runcomplete.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/bin/modulediff.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/bin/moduleOccupancyPlots.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/bin/moduleOccupancyTrend.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-23-2300/src/DQM/SiStripMonitorClient/src/SiStripDcsInfo.cc:338:33: error: no viable conversion from 'map<[...], uint32_t>' to 'map<[...], uint16_t>'
    std::map<uint32_t,uint16_t> lumiCountBadModules = it->second.NLumiDetectorIsFaulty;
                                ^                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/nweek-02464/slc6_amd64_gcc530/external/gcc/5.3.0/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../include/c++/5.3.0/bits/stl_map.h:185:7: note: candidate constructor not viable: no known conversion from 'std::map<uint32_t, uint32_t>' (aka 'map<unsigned int, unsigned int>') to 'const std::map<unsigned int, unsigned short, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, unsigned short> > > &' for 1st argument
      map(const map& __x)
      ^
</pre>

--- a/DQM/SiStripMonitorClient/interface/SiStripDcsInfo.h
+++ b/DQM/SiStripMonitorClient/interface/SiStripDcsInfo.h
@@ -32,6 +32,7 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <unordered_map>
 
 class DQMStore;
 class MonitorElement;
@@ -86,7 +87,7 @@ private:
     MonitorElement* DcsFractionME;
     int TotalDetectors;
     std::vector<uint32_t> FaultyDetectors;
-    std::map<uint32_t,uint16_t> NLumiDetectorIsFaulty;
+    std::unordered_map<uint32_t,uint16_t> NLumiDetectorIsFaulty;
   };
 
   std::map <std::string, SubDetMEs> SubDetMEsMap;

--- a/DQM/SiStripMonitorClient/interface/SiStripDcsInfo.h
+++ b/DQM/SiStripMonitorClient/interface/SiStripDcsInfo.h
@@ -86,7 +86,7 @@ private:
     MonitorElement* DcsFractionME;
     int TotalDetectors;
     std::vector<uint32_t> FaultyDetectors;
-    std::map<uint32_t,uint32_t> NLumiDetectorIsFaulty;
+    std::map<uint32_t,uint16_t> NLumiDetectorIsFaulty;
   };
 
   std::map <std::string, SubDetMEs> SubDetMEsMap;

--- a/DQM/SiStripMonitorClient/src/SiStripDcsInfo.cc
+++ b/DQM/SiStripMonitorClient/src/SiStripDcsInfo.cc
@@ -335,8 +335,8 @@ void SiStripDcsInfo::addBadModules() {
 
   for (std::map<std::string, SubDetMEs>::iterator it = SubDetMEsMap.begin(); it != SubDetMEsMap.end(); it++) {
 
-    std::map<uint32_t,uint16_t> lumiCountBadModules = it->second.NLumiDetectorIsFaulty;
-    for(std::map<uint32_t,uint16_t>::iterator ilumibad = lumiCountBadModules.begin(); 
+    std::unordered_map<uint32_t,uint16_t> lumiCountBadModules = it->second.NLumiDetectorIsFaulty;
+    for(std::unordered_map<uint32_t,uint16_t>::iterator ilumibad = lumiCountBadModules.begin(); 
         ilumibad != lumiCountBadModules.end(); ilumibad++) {
       uint32_t ibad = (*ilumibad).first;
       uint32_t nBadLumi = (*ilumibad).second;

